src/AuthProvider.tsx

```diff
@@ -81,10 +95,27 @@ export const AuthProvider = (props: AuthProviderProps): JSX.Element => {
         ...userManagerProps
     } = props;
 
-    const [userManager] = React.useState<UserManager>(() => new UserManager(userManagerProps));
+    const [userManager] = React.useState(() =>
+        typeof window === "undefined" ? null : new UserManager(userManagerProps)
+    );
     const [state, dispatch] = React.useReducer(reducer, initialAuthState);
+    const userManagerContext = React.useMemo(
+        () => ({
+            settings: userManager ?? {},
```

Comment:
only the settings are passed, we need something like:
`settings: userManager?.settings ?? {},`
or even compatible when we switch to oidc-client-ts:
`settings: userManager?.settings ?? {...userManagerProps},`

Reply:
its might be even easier if we always create the manager. Like this we can make use of userManager.settings in userManagerContext.
And instead to check for userManager === null, we could make a new helper variable:
`const isSSR = typeof window === "undefined";` and check for that...

Reply:
Thanks for catching this, it was a typo.

Also, UserManager class will need some changes to prevent it from reaching out to the metadata endpoint for the `authority` passed to it if we want to allow constructing it server-side.

Reply:
Actually, I'm reviewing the code in oidc-client, and it looks like maybe we just need to disable the session monitor in server contexts by default? I tried disabling it before I put together this PR and couldn't get it to work. I'm not familiar enough with oidc-client's internals to know which parts are server-compatible vs which parts aren't.

Reply:
ok the oidc-client-ts fork has a bigger issue where not even the settings store can be instantiated server-side due to the [use of `sessionStorage`](https://github.com/pamapa/oidc-client-ts/blob/e563b22585e623236b4bd596822d56a62c4806d5/src/UserManagerSettings.ts#L99): I get `ReferenceError: sessionStorage is not defined` when I allow the manager to be created. I think the only good short-term solution is just to avoid constructing a UserManager in this instance.

Reply:
session monitor was enabled i disabled it recently by default